### ISSUE TITLE
Add a spell checker

### ISSUE THREAD

**Issue by ganthern** [tutao/tutanota#1069]

**Comment by ganthern** [comment]
This is going to be a bigger project. Most spell check implementations for electron request dictionary files from google or similar and we can't have that. Also, they seem to require native dependencies.

**Comment by pczajkowski** [comment]
It would be really nice to have though.

**Comment by ganthern** [comment]
[node-spellchecker](https://github.com/atom/node-spellchecker) with [webFrame.setSpellCheckProvider](https://www.electronjs.org/docs/api/web-frame#webframesetspellcheckproviderlanguage-provider) may be a solution.

**Comment by vladimiry** [comment]
`webFrame.setSpellCheckProvider` is like interface that requires external library for working with dictionaries, like `node-spellchecker`. This combination works well in https://github.com/vladimiry/ElectronMail.

But Electron recently added support for chromium-builtin spellchecker functionality, see https://github.com/electron/electron/pull/20692 and https://github.com/electron/electron/blob/v8.0.0-beta.8/docs/api/session.md#sessetspellcheckerlanguageslanguages It's in beta state for now though but Electron v8 is going to be released quite soon.

By the way, I would would not recommend using https://github.com/electron-userland/electron-spellchecker as it's messy (built on `webFrame.setSpellCheckProvider` + `node-spellchecker`).

**Comment by charlag** [comment]
@vladimiry thanks for pointing out. We already found about about `node-spellchecker` and the killer feature is that you don't have to download dictionaries for it, as far as I understand. With new Electron API you have to download them from Chromium or host yourself, as far as I understand. It should be more reliable tho.

**Comment by vladimiry** [comment]
> you don't have to download dictionaries for it

In my experience, it depends on the system and the package type the app is distributed in. For Snap package type I had to embed the dictionaries into the `*.snap` app package as Snap packages come with builtin isolation capabilities that restrict proper resolving system-wide dictionaries. Besides on Window 7 dictionaries don't get resolved well, so I had to embed dictionaries into the `.exe` package too.
